frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segments_view.tsx

```diff
@@ -396,16 +409,41 @@ class SegmentsView extends React.Component<Props, State> {
         searchableTreeItemList.push(item);
       });
 
-      return {
+      updateStateObject = {
         groupTree: generatedGroupTree,
         searchableTreeItemList,
         prevProps: nextProps,
       };
-    } else {
+    }
+    if (prevState.prevProps?.isosurfaces !== isosurfaces) {
```

Comment:
I am not sure I understand the reason for this code. Can you elaborate please. How can I test this?

Reply:
This code makes sure that changing the visibility of a mesh (e.g. by clicking the checkbox) gets noticed by the segments_view as the visibility is part of the isosurface. 
This is needed because I toggle the "Hide/Show meshes" option based on the mesh visibility in the group. 
You can test it by loading all meshes in a group. Now, the context menu should say "Hide...". If you deselect any mesh's  visibility checkbox, the context menu notices and will then read "Show all meshes".
Does that make sense?

Reply:
also see philipps comment for this: https://github.com/scalableminds/webknossos/pull/7164#pullrequestreview-1507178411